RELEASE/scripts/sl_ascend.ash

```diff
@@ -14008,7 +14082,7 @@ boolean L3_tavern()
 	{
 		return false;
 	}
-	if(my_adventures() < 5)
+	if(my_adventures() < 10)
```

Comment:
You could count the number of explored tiles elements in the tavern by counting the number of non-zero elements in `prefref tavernLayout`. You'll need to hit `cellar.php` first, though.

(my worry is we start the tavern with 14 adventures, adventure 5 times, and then give up and go somewhere else while wearing all of our expensive buffs)

Reply:
My worry about that approach is 2 fold:
1) If someone manually mucked around in there any non-zero values could be irrelevant to our intended path.
2) I don't know how to do that. (though I could mess around and figure it out maybe...)

Reply:
The current code, if we start the basement with 14 adventures left, will leave early and waste those buff turns somewhere else. This seems like a serious bug to me, since it makes it difficult to reason about when the tavern will be completed. I'm OK with a bug if someone has manually wandered around the tavern, since the *worst possible case* there is that we run out of adventures at the end of the day and have to pick it up tomorrow. Please fix it.

2. Sample code for dealing with strings: `string s = "00001"; for (int i=0; i < length(s); i++) { print(s.char_at(i));}`

Reply:
Thanks, I added it to my notes.